### PR TITLE
Fix #719: Update UI and Strings for Sync.

### LINKED ISSUE #719: Sync UI Updates
Update Sync UI and strings
sync
[sync_ios.zip](https://github.com/brave/brave-ios/files/2757614/sync_ios.zip)

Specs - 
[Mobile Sync Updates.pdf](https://github.com/brave/brave-ios/files/2770209/Mobile.Sync.Updates.pdf)

### PR DESCRIPTION
**Pull request by iccub**
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-ios/issues) for my issue if one did not already exist.
- [x] My patch or PR title has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!` (or `No Bug: <message>` if no relevant ticket)
- [ ] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [ ] New files have MPL-2.0 license header.

## Test Plan:

Look at the pdf from https://github.com/brave/brave-ios/issues/719
And go through all Sync screens and verify that it works more or less similar to it.
Try to catch any typos, as I have written the copy by hand

### Screenshots:

zrzut ekranu 2019-01-18 o 13 41 12
zrzut ekranu 2019-01-18 o 13 43 35
zrzut ekranu 2019-01-18 o 13 43 42
zrzut ekranu 2019-01-18 o 13 43 54
zrzut ekranu 2019-01-18 o 15 23 40
zrzut ekranu 2019-01-18 o 15 23 43
zrzut ekranu 2019-01-18 o 15 23 46
zrzut ekranu 2019-01-18 o 15 23 51
zrzut ekranu 2019-01-18 o 15 23 59
zrzut ekranu 2019-01-18 o 15 24 01

## Reviewer Checklist:

- [x] PR is linked to an issue via [Zenhub](https://www.zenhub.com/extension).
- [x] Issues are assigned to at least one epic.
- [x] Issues include necessary QA labels:
  - [x] `QA/(Yes|No)`
  - [x] `release-notes/(include|exclude)`
  - [x] `bug` / `enhancement`
- [ ] Necessary security reviews have taken place.
- [ ] Adequate test coverage exists to prevent regressions.
- [x] Adaquate test plan exists for QA to validate (if applicable)

### CODE REVIEW

File: Client/Frontend/Sync/SyncPairWordsViewController.swift [hunk 1]
```diff
@@ -29,6 +29,16 @@ class SyncPairWordsViewController: SyncViewController {
         return button
     }()
     
+    lazy var useCameraButton: UIButton = {
```
Comment (jhreis) [inline Client/Frontend/Sync/SyncPairWordsViewController.swift]: Lazy not needed here.

File: Client/Frontend/Sync/SyncSettingsTableViewController.swift [hunk 2]
```diff
@@ -89,23 +98,82 @@ class SyncSettingsTableViewController: UITableViewController {
...
+        
+        let removeAction = UIAlertAction(title: Strings.SyncRemoveDeviceAction, style: .destructive) { _ in
+            var alertType: DeviceRemovalType?
+            
+            if deviceCount == 1 {
+                alertType = .lastDeviceLeft
+            } else if device.isCurrentDevice {
+                alertType = .currentDevice
+            } else {
+                alertType = .otherDevice
```
Comment (jhreis) [inline Client/Frontend/Sync/SyncSettingsTableViewController.swift]: I think a bit cleaner here is not using `else` but instead setting alertType explicitly:

`var alertType = DeviceRemovalType.otherDevice // notice, no optional`

Then overwrite via the if-else.

This way it is also not optional so you don't need to unwrap it then.

File: Client/Frontend/Sync/SyncSettingsTableViewController.swift [hunk 3]
```diff
@@ -89,23 +98,82 @@ class SyncSettingsTableViewController: UITableViewController {
...
+            title = String(format: Strings.SyncRemoveCurrentDeviceTitle, "\(deviceName) (\(Strings.SyncThisDevice))")
+            message = Strings.SyncRemoveCurrentDeviceMessage
+            removeButtonName = Strings.RemoveDevice
+        case .otherDevice:
+            title = String(format: Strings.SyncRemoveOtherDeviceTitle, deviceName)
+            message = Strings.SyncRemoveOtherDeviceMessage
+            removeButtonName = Strings.RemoveDevice
+        }
+        
+        guard let popupTitle = title, let popupMessage = message, let popupButtonName = removeButtonName else { fatalError() }
```
Comment (jhreis) [inline Client/Frontend/Sync/SyncSettingsTableViewController.swift:154]: Good job with `fatalError()`, clean.

File: Client/Frontend/Sync/SyncSettingsTableViewController.swift [hunk 4]
```diff
@@ -89,23 +98,82 @@ class SyncSettingsTableViewController: UITableViewController {
...
+        
+        popup.addButton(title: Strings.CancelButtonTitle, fontSize: fontSize) { return .flyDown }
+        popup.addButton(title: popupButtonName, type: .destructive, fontSize: fontSize) {
+            switch type {
+            case .lastDeviceLeft, .currentDevice:
+                Sync.shared.leaveSyncGroup()
+                self.navigationController?.popToRootViewController(animated: true)
+            case .otherDevice:
+                device.remove()
+                break
```
Comment (jhreis) [inline Client/Frontend/Sync/SyncSettingsTableViewController.swift]: Is `break` needed?

File: Client/Frontend/Sync/SyncSettingsTableViewController.swift [hunk 5]
```diff
@@ -115,7 +183,7 @@ class SyncSettingsTableViewController: UITableViewController {
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
         let deviceCount = frc?.fetchedObjects?.count ?? 0
         
-        return section == Sections.buttons.rawValue ? ButtonsSection.count : deviceCount
+        return section == Sections.buttons.rawValue ? 1 : deviceCount
```
Comment (jhreis) [inline Client/Frontend/Sync/SyncSettingsTableViewController.swift]: Can you add a comment on this?

**Review by jhreis — CHANGES_REQUESTED** [review]

File: Client/Frontend/Sync/SyncSettingsTableViewController.swift (hunk 5 above)
Comment (iccub) [reply Client/Frontend/Sync/SyncSettingsTableViewController.swift]: `remove device` button is gone, and we are left with one button so I decided to drop the buttons `enum` altogether.

File: Client/Frontend/Sync/SyncSettingsTableViewController.swift [hunk 6]
```diff
@@ -114,8 +177,9 @@ class SyncSettingsTableViewController: UITableViewController {
     
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
         let deviceCount = frc?.fetchedObjects?.count ?? 0
+        let buttonsCount = 1
         
-        return section == Sections.buttons.rawValue ? ButtonsSection.count : deviceCount
+        return section == Sections.buttons.rawValue ? buttonsCount : deviceCount
```
Comment (jhreis) [inline Client/Frontend/Sync/SyncSettingsTableViewController.swift:182]: Just to clarify, I just meant a normal comment, but I see what's going on here. No need for the constant here, I think previously was clear enough. But I understand better what is going on. 

Not going to block PR on this. Just wanted to leave a comment.

**Review by jhreis — APPROVED** [review]

**Pull request merged**